### PR TITLE
OKP Test Utilities

### PR DESCRIPTION
**Pull request by OliverPearmain-Pusher**
### What?
Added `Test Utilities` target. 
Moved `TestLogger` into this target (from Unit Tests)

### Why?
Allows us to write code once that can be shared between all test targets.

----

### CODE REVIEW

File: Chatkit.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1586,6 +1686,30 @@
...
 		};
+		73E585C423E087500047C6FA /* Debug */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				DYLIB_COMPATIBILITY_VERSION = 1;
+				DYLIB_CURRENT_VERSION = 1;
+				DYLIB_INSTALL_NAME_BASE = "@rpath";
+				INFOPLIST_FILE = "Test Utilities/Supporting Files/Info.plist";
+				PRODUCT_BUNDLE_IDENTIFIER = "com.pusher.chatkit.$(PRODUCT_NAME:rfc1034identifier)";
+				PRODUCT_NAME = "$(TARGET_NAME:c99extidentifier)";
```
Comment (grzesiekko) [inline Chatkit.xcodeproj/project.pbxproj]:
```suggestion
				PRODUCT_NAME = TestUtilities;
```

File: Chatkit.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1586,6 +1686,30 @@
...
+		};
+		73E585C523E087500047C6FA /* Release */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				DYLIB_COMPATIBILITY_VERSION = 1;
+				DYLIB_CURRENT_VERSION = 1;
+				DYLIB_INSTALL_NAME_BASE = "@rpath";
+				INFOPLIST_FILE = "Test Utilities/Supporting Files/Info.plist";
+				PRODUCT_BUNDLE_IDENTIFIER = "com.pusher.chatkit.$(PRODUCT_NAME:rfc1034identifier)";
+				PRODUCT_NAME = "$(TARGET_NAME:c99extidentifier)";
```
Comment (grzesiekko) [inline Chatkit.xcodeproj/project.pbxproj]:
```suggestion
				PRODUCT_NAME = TestUtilities;
```

File: Test Utilities/Supporting Files/Test_Utilities.h [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#import <Foundation/Foundation.h>
```
Comment (grzesiekko) [inline Test Utilities/Supporting Files/Test_Utilities.h]:
```suggestion
@import Foundation;
```

File: Test Utilities/Supporting Files/Test_Utilities.h [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+//! Project version number for Test_Utilities.
+FOUNDATION_EXPORT double Test_UtilitiesVersionNumber;
```
Comment (grzesiekko) [inline Test Utilities/Supporting Files/Test_Utilities.h]:
```suggestion
FOUNDATION_EXPORT double TestUtilitiesVersionNumber;
```

File: Test Utilities/Supporting Files/Test_Utilities.h [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+//! Project version string for Test_Utilities.
+FOUNDATION_EXPORT const unsigned char Test_UtilitiesVersionString[];
```
Comment (grzesiekko) [inline Test Utilities/Supporting Files/Test_Utilities.h]:
```suggestion
FOUNDATION_EXPORT const unsigned char TestUtilitiesVersionString[];
```

File: Test Utilities/Supporting Files/Test_Utilities.h [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+// In this header, you should import all the public headers of your framework using statements like #import <Test_Utilities/PublicHeader.h>
+
+
```
Comment (grzesiekko) [inline Test Utilities/Supporting Files/Test_Utilities.h]: Assuming that we want to name the product `TestUtilities`, this file should be renamed to `TestUtilities.h`.

File: Unit Tests/Tests/Authentication/Token Providers/TestTokenProviderTests.swift [hunk 7]
```diff
@@ -1,6 +1,7 @@
 import XCTest
 import Mockingjay
 import PusherPlatform
+@testable import Tests_Utilities
```
Comment (grzesiekko) [inline Unit Tests/Tests/Authentication/Token Providers/TestTokenProviderTests.swift]:
```suggestion
@testable import TestsUtilities
```

File: Unit Tests/Tests/Chatkit/ChatkitTests.swift [hunk 8]
```diff
@@ -1,4 +1,5 @@
 import XCTest
+@testable import Tests_Utilities
```
Comment (grzesiekko) [inline Unit Tests/Tests/Chatkit/ChatkitTests.swift]:
```suggestion
@testable import TestsUtilities
```

File: Unit Tests/Tests/Networking/Controller/NetworkingControllerTests.swift [hunk 9]
```diff
@@ -1,4 +1,5 @@
 import XCTest
+@testable import Tests_Utilities
```
Comment (grzesiekko) [inline Unit Tests/Tests/Networking/Controller/NetworkingControllerTests.swift]:
```suggestion
@testable import TestsUtilities
```

File: Unit Tests/Tests/Networking/Services/ChatServiceTests.swift [hunk 10]
```diff
@@ -1,5 +1,6 @@
 import XCTest
 import PusherPlatform
+@testable import Tests_Utilities
```
Comment (grzesiekko) [inline Unit Tests/Tests/Networking/Services/ChatServiceTests.swift]:
```suggestion
@testable import TestsUtilities
```

**Comment by grzesiekko** [comment]
It looks like unit tests do not compile anymore as there is no such module as `Tests_Utilities`.

**Comment by grzesiekko** [comment]
It would be also great to get rid of the warning about Swift 5 migration for the new target:

```
LastSwiftMigration = 1130
```

**Comment by OliverPearmain-Pusher** [comment]
Checking now.  I did run the tests and they passed but perhaps it was using some cached artifacts.

**Comment by OliverPearmain-Pusher** [comment]
@grzesiekko think I've addressed all your comments

**Pull request merged**